lib/matplotlib/axes/_base.py

```diff
@@ -229,24 +225,76 @@ def _xy_from_xy(self, x, y):
             y = y[:, np.newaxis]
         return x, y
 
+    def _getdefaults(self, *kwargs):
+        """
+        Only advance the cycler if the cycler has information that
+        is not specified in any of the supplied tuple of dicts
+
+        Returns a dictionary of defaults if there are any
+        keys that are not found in any of the supplied dictionaries.
+        If the supplied dictionaries have non-None values for
+        everything the property cycler has, then just return
+        an empty dictionary.
+
+        """
+        if any([all([kw.get(k, None) is None for kw in kwargs])
+               for k in self._prop_keys]):
+            default_dict = six.next(self.prop_cycler)
```

Comment:
Something is wrong with the above if-statement. I just tried adding a test where the cycler should not have been advanced, and it did get advanced. I suspect something is wrong with my compound for-statements...